### PR TITLE
Allow `package(set)` at the top level decl site

### PR DESCRIPTION
**Pull request by elsh**
-Allow `package(set)` at the top level decl site
-Treat `package` as a non-contextual keyword when applicable 
-Add tests
Resolves rdar://105978922, rdar://104931420

### CODE REVIEW

File: test/Sema/accessibility_package_contextual_keyword.swift [hunk 1]
```diff
@@ -0,0 +1,64 @@
+// RUN: %empty-directory(%t)
+// RUN: %{python} %utils/split_file.py -o %t %s
```
Comment (xymus) [inline test/Sema/accessibility_package_contextual_keyword.swift:2]: General comment, this is the old version of the tool. For new files I'd recommend to use the following which is a bit cleaner:
```
// RUN: split-file %s %t
//--- File.swift
```

File: test/Sema/accessibility_package_contextual_keyword.swift [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+// RUN: %target-swift-frontend-typecheck -verify -disable-availability-checking %t/B.swift -package-name myPkg
+// RUN: %target-swift-frontend-typecheck -verify -disable-availability-checking %t/D.swift -package-name myPkg
+// RUN: not %target-swift-frontend-typecheck -verify -disable-availability-checking %t/C.swift -package-name myPkg 2>&1 | %FileCheck %s
+
+// BEGIN main.swift
+package(set) public var a: String // should pass when `package` modifier is used at top level decls
+public package(set) var b: String
+package let c: Int
+package var d: Int
+package func f() {}
```
Comment (xymus) [inline test/Sema/accessibility_package_contextual_keyword.swift:15]: Would calling a function named `package` still be accepted at the top level?

```
func package() {}
package()
```

Comment (elsh) [reply test/Sema/accessibility_package_contextual_keyword.swift:15]: yes it works. I'll add them to the test

File: test/Sema/accessibility_package_contextual_keyword.swift (hunk 1 above)
Comment (elsh) [reply test/Sema/accessibility_package_contextual_keyword.swift:2]: sg

**Comment by elsh** [comment]
@swift-ci smoke test

**Review by xymus — APPROVED** [review]

**Comment by elsh** [comment]
@swift-ci smoke test

**Pull request merged**